plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/LeasedPlcConnection.java

```diff
@@ -66,7 +76,7 @@ public void connect() throws PlcConnectionException {
     @Override
     public boolean isConnected() {
         if(connection == null) {
-            throw new PlcRuntimeException("Error using leased connection after returning it to the cache.");
+            return false;
```

Comment:
Not quite sure this is the best way to do this. Because for me there's a difference, if we're not connected or if we already gave back the handle. The way I implemented it gives the user the explicit feedback, that he's doing something bad. If we return false, he might think he just needs to re-connect.

Reply:
If we lose the connection we can handle reconnection

Reply:
By the way. I'm working on eclipse kura. It required working uninterrupted. We must explicit reconect signal to reconnect not a exception.

Reply:
> 



> 

I have encountered a scenario where the PLC connection appeared to be disconnected, but in reality it was still connected due to certain reasons. However, we kept trying to reconnect and as a result, the TCP estab value increased to tens of thousands. This caused the PLC to be disconnected and also resulted in server network issues. Therefore, I'm looking to see if there is a better solution available.

Reply:
> I have already explained that previous PR are the basis for subsequent PRs. Next I want to submit a PR for PL4x

First of all, thank you for your PR. Secondly, I think it would be beneficial if you could discuss your entire design approach with everyone, rather than just submitting a basic PR and following up with subsequent PRs.

Reply:
I still don't quite understand the problem with this ... if you returned the connection to the pool, you should not do anything with it ... that's sort of the contract on using the pool. If there's some logic calling "isConnected()" after it has been returned, that's just a flaw in the program using it. I still think we should continue to throw an exception here.

Reply:
We use PLC4x to collect data for long term work. If the connection is closed abnormally, I don’t want to always use the try catch method. If we use this to detect that the connection is closed, we only need to borrow it again.

Reply:
I would say that's a usecase particular to how you use PLC4X, however out API shouldn't be optimized towards one particular usecase. I think try-catch-blocks don't consume any noticable amount of resources. So I would like to ask to to stick to using try-catch blocks and us keeping the API coherent.

Reply:
Just connect is not allowed. I think isConeccted is should reserve the functionality of the original API

Reply:
I'm wondering why you want to disable this API functionality and replace it with an error